src/slowmokit/methods/metrics/silhouette_score.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file methods/metrics/silhouette_score.cpp
+ *
+ * Implementation of the Silhouette Score main program
+ */
+
+#include "silhouette_score.hpp"
+template<class T>
+double SilhouetteScore(std::vector<std::vector<T>> x, std::vector<int> y,
+                       int numClusters, std::string typeDist)
+{
+  std::vector<std::vector<double>> distances(x.size(),
+                                             std::vector<double>(x.size()));
+  for (int i = 0; i < x.size(); i++)
+  { // iterating over each x[i]
+    for (int j = 0; j < x.size(); j++)
+    { // iterating over each x[i]
+      if (i == j)
+      {
+        distances[i][j] = 0.0; // initialize distance=0 if computing distance
+                               // between same values
+      }
+      else
+      {
+        if (typeDist == "euclidean")
```

Comment:
- First convert typeDist to all lower character and then compare it with `euclidean` so even if user passes `Euclidean` it works.

<details>
<summary> How to make a string in lower_case in cpp? </summary>

```cpp
std::string data = "Abc";
std::transform(data.begin(), data.end(), data.begin(),
    [](unsigned char c){ return std::tolower(c); });
```
</details>